lib/iex/lib/iex.ex

```diff
@@ -470,14 +505,130 @@ defmodule IEx do
+    * `IEx.Helpers.reset_break/3` - resets all breaks for the given module, function, arity
+    * `IEx.Helpers.respawn/0` - starts a new shell (breakpoints will ask for permission once more)
+    * `IEx.Helpers.whereami/1` - shows the current location
+
+  By default, the number of stops in a breakpoint is 1. Any follow-up
+  call won't stop the code execution unless another breakpoint is set.
+
+  Alternatively, the number of be increased by passing the `stops`
+  argument. `IEx.Helpers.reset_break/1` and `IEx.Helpers.reset_break/3`
+  can be used to reset the number back to zero. Note the module remains
+  "instrumented" even after all stops on all breakpoints are consumed.
+  You can remove the instrumentation in a given module by calling
+  `IEx.Helpers.remove_breaks/1` and on all modules by calling
+  `IEx.Helpers.remove_breaks/0`.
+
+  To exit a breakpoint, the developer can either invoke `continue()`,
+  which will block the shell until the next breakpoint is found or
+  the process terminates, or invoke `respawn()`, which starts a new IEx
+  shell, freeing up the pried one.
+
+  This functionality only works on Elixir code and requires OTP 20+.
+
+  ## Examples
+
+  The following sets up a breakpoint on `URI.decode_query/2`:
+
+      IEx.break!(URI, :decode_query, 2)
+
+  The following call will setup a breakpoint that stops once.
+  To set a breakpoint that will stop 10 times:
+
+      IEx.break!(URI, :decode_query, 10)
+
+  `IEx.break!/2` is a convenience macro that allows breakpoints
+  to be given in the `Mod.fun/arity` format:
+
+      require IEx
+      IEx.break!(URI.decode_query/2)
+
+  Or to set a breakpoint that will stop 10 times:
+
+      IEx.break!(URI.decode_query/2, 10)
+
+  This function returns the breakpoint ID and will raise if there
+  is an error setting up the breakpoint.
+  """
+  def break!(module, function, arity, stops \\ 10) do
+    case IEx.Pry.break(module, function, arity, stops) do
+      {:ok, id} ->
+        id
+      {:error, kind} ->
+        message =
+          case kind do
+            :missing_debug_info ->
+              "module #{inspect module} was not compiled with debug_info"
+            :no_beam_file ->
+              "could not find .beam file for #{inspect module}"
+            :non_elixir_module ->
+              "module #{inspect module} was not written in Elixir"
+            :otp_20_is_required ->
+              "you are running on an earlier OTP version than OTP 20"
+            :outdated_debug_info ->
+              "module #{inspect module} was not compiled with the latest debug_info"
+            :recompilation_failed ->
+              "the module could not be compiled with breakpoints (likely an internal error)"
+            :unknown_function_arity ->
+              "unknown function/macro #{Exception.format_mfa(module, function, arity)}"
+          end
+        raise ArgumentError, "could not set breakpoint, " <> message
```

Comment:
Is this really an `ArgumentError`? It feels like most of the errors above are not, for example the `:otp_20_is_required`. What if we introduce a `IEx.BreakError` or something similar?